amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/DeclaredAnnotationSchemaValidator.scala

```diff
@@ -0,0 +1,49 @@
+package amf.apicontract.internal.spec.common.parser
+
+import amf.aml.internal.semantic.{AnnotationSchemaValidator, IgnoreAnnotationSchemaValidator}
+import amf.apicontract.internal.validation.definitions.ParserSideValidations.{
+  AnnotationSchemaMustBeAny,
+  MissingAnnotationSchema
+}
+import amf.core.client.scala.errorhandling.AMFErrorHandler
+import amf.core.client.scala.model.domain.extensions.CustomDomainProperty
+import amf.shapes.client.scala.model.domain.AnyShape
+import amf.shapes.internal.domain.metamodel.AnyShapeModel
+import org.mulesoft.lexer.SourceLocation
+import org.yaml.model.YNode
+
+case class DeclaredAnnotationSchemaValidator(override val annotationIndex: Map[String, CustomDomainProperty])
+    extends AnnotationSchemaValidator {
+  override def validate(name: String, key: YNode, eh: AMFErrorHandler): Unit = {
+    annotationIndex.get(name).fold(addMissingAnnotationViolation(name, key, eh))(checkAnnotationSchema(_, eh))
+  }
+
+  private def checkAnnotationSchema(annotation: CustomDomainProperty, eh: AMFErrorHandler): Unit = {
+    annotation.schema match {
+      case any: AnyShape if any.meta != AnyShapeModel =>
```

Comment:
there is an auxliary method in AnyShape for this check

Reply:
If you are talking about isAnyType of isNotExplicit isn't the same check.
The second one checks if there is a "type" key written, and is not my case.